<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Engine Monitoring System</string>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::TabShape::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QWidget" name="dataWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QTabWidget" name="tabWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>690</width>
           <height>0</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>1</number>
         </property>
         <widget class="QWidget" name="tableModeTab">
          <attribute name="title">
           <string>Tables</string>
          </attribute>
          <widget class="QTableWidget" name="sensorTable">
           <property name="geometry">
            <rect>
             <x>450</x>
             <y>10</y>
             <width>230</width>
             <height>480</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <column>
            <property name="text">
             <string>Sensor</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Status</string>
            </property>
           </column>
           <item row="0" column="0">
            <property name="text">
             <string>Oil Pressure</string>
            </property>
           </item>
           <item row="1" column="0">
            <property name="text">
             <string>Oil Temp</string>
            </property>
           </item>
           <item row="2" column="0">
            <property name="text">
             <string>Fuel Flow</string>
            </property>
           </item>
           <item row="3" column="0">
            <property name="text">
             <string>Fuel</string>
            </property>
           </item>
           <item row="4" column="0">
            <property name="text">
             <string>EGT</string>
            </property>
           </item>
           <item row="5" column="0">
            <property name="text">
             <string>Torque</string>
            </property>
           </item>
           <item row="6" column="0">
            <property name="text">
             <string>Indicated Power</string>
            </property>
           </item>
           <item row="7" column="0">
            <property name="text">
             <string>Frictional Power</string>
            </property>
           </item>
           <item row="8" column="0">
            <property name="text">
             <string>Therm Efficiency</string>
            </property>
           </item>
           <item row="9" column="0">
            <property name="text">
             <string>Air-Fuel Ratio</string>
            </property>
           </item>
           <item row="10" column="0">
            <property name="text">
             <string>Motor Speed</string>
            </property>
           </item>
           <item row="11" column="0">
            <property name="text">
             <string>Output Air Speed</string>
            </property>
           </item>
           <item row="12" column="0">
            <property name="text">
             <string>Vibration</string>
            </property>
           </item>
           <item row="13" column="0">
            <property name="text">
             <string>Body Temp</string>
            </property>
           </item>
           <item row="14" column="0">
            <property name="text">
             <string>Air Temp</string>
            </property>
           </item>
          </widget>
          <widget class="QTableWidget" name="dataTable">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>11</x>
             <y>11</y>
             <width>430</width>
             <height>480</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <row/>
           <column>
            <property name="text">
             <string>Data</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Min Value</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Max Value</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
           <item row="0" column="0">
            <property name="text">
             <string>Oil Pressure</string>
            </property>
           </item>
           <item row="0" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="0" column="2">
            <property name="text">
             <string>1000</string>
            </property>
           </item>
           <item row="1" column="0">
            <property name="text">
             <string>Oil Temp</string>
            </property>
           </item>
           <item row="1" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="1" column="2">
            <property name="text">
             <string>400</string>
            </property>
           </item>
           <item row="2" column="0">
            <property name="text">
             <string>Fuel Flow</string>
            </property>
           </item>
           <item row="2" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="2" column="2">
            <property name="text">
             <string>800</string>
            </property>
           </item>
           <item row="3" column="0">
            <property name="text">
             <string>Fuel</string>
            </property>
           </item>
           <item row="3" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="3" column="2">
            <property name="text">
             <string>800</string>
            </property>
           </item>
           <item row="4" column="0">
            <property name="text">
             <string>EGT</string>
            </property>
           </item>
           <item row="4" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="4" column="2">
            <property name="text">
             <string>400</string>
            </property>
           </item>
           <item row="5" column="0">
            <property name="text">
             <string>Torque</string>
            </property>
           </item>
           <item row="5" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="5" column="2">
            <property name="text">
             <string>400</string>
            </property>
           </item>
           <item row="6" column="0">
            <property name="text">
             <string>Indicated Power</string>
            </property>
           </item>
           <item row="6" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="6" column="2">
            <property name="text">
             <string>400</string>
            </property>
           </item>
           <item row="7" column="0">
            <property name="text">
             <string>Frictional Power</string>
            </property>
           </item>
           <item row="7" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="7" column="2">
            <property name="text">
             <string>400</string>
            </property>
           </item>
           <item row="8" column="0">
            <property name="text">
             <string>Therm Efficiency</string>
            </property>
           </item>
           <item row="8" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="8" column="2">
            <property name="text">
             <string>100</string>
            </property>
           </item>
           <item row="9" column="0">
            <property name="text">
             <string>Air-Fuel Ratio</string>
            </property>
           </item>
           <item row="9" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="9" column="2">
            <property name="text">
             <string>20</string>
            </property>
           </item>
           <item row="10" column="0">
            <property name="text">
             <string>Motor Speed</string>
            </property>
           </item>
           <item row="10" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="10" column="2">
            <property name="text">
             <string>1000</string>
            </property>
           </item>
           <item row="11" column="0">
            <property name="text">
             <string>Output Air Speed</string>
            </property>
           </item>
           <item row="11" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="11" column="2">
            <property name="text">
             <string>1000</string>
            </property>
           </item>
           <item row="12" column="0">
            <property name="text">
             <string>Vibration</string>
            </property>
           </item>
           <item row="12" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="12" column="2">
            <property name="text">
             <string>100</string>
            </property>
           </item>
           <item row="13" column="0">
            <property name="text">
             <string>Body Temp</string>
            </property>
           </item>
           <item row="13" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="13" column="2">
            <property name="text">
             <string>400</string>
            </property>
           </item>
           <item row="14" column="0">
            <property name="text">
             <string>Air Temp</string>
            </property>
           </item>
           <item row="14" column="1">
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item row="14" column="2">
            <property name="text">
             <string>400</string>
            </property>
           </item>
          </widget>
         </widget>
         <widget class="QWidget" name="settingsTab">
          <attribute name="title">
           <string>Settings</string>
          </attribute>
          <widget class="QLabel" name="statusLabel">
           <property name="geometry">
            <rect>
             <x>100</x>
             <y>250</y>
             <width>140</width>
             <height>30</height>
            </rect>
           </property>
           <property name="text">
            <string>Status: Disconnected</string>
           </property>
          </widget>
          <widget class="QComboBox" name="portComboBox">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>10</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="placeholderText">
            <string>Select Port</string>
           </property>
           <item>
            <property name="text">
             <string>Select</string>
            </property>
           </item>
          </widget>
          <widget class="QPushButton" name="startButton">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>210</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
          <widget class="QPushButton" name="stopButton">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>210</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
          <widget class="QComboBox" name="baudRateComboBox">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>50</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="placeholderText">
            <string>Select Port</string>
           </property>
           <item>
            <property name="text">
             <string>Select</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4800</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>19200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>38400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>57600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>115200</string>
            </property>
           </item>
          </widget>
          <widget class="QComboBox" name="parityComboBox">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>90</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="placeholderText">
            <string>Select Port</string>
           </property>
           <item>
            <property name="text">
             <string>Select</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>No parity</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Odd</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Even</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Mark</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Space</string>
            </property>
           </item>
          </widget>
          <widget class="QComboBox" name="stopBitComboBox">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>130</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="placeholderText">
            <string>Select Port</string>
           </property>
           <item>
            <property name="text">
             <string>Select</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1.5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
          </widget>
          <widget class="QLabel" name="portLabel">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>10</y>
             <width>30</width>
             <height>30</height>
            </rect>
           </property>
           <property name="text">
            <string>Port:</string>
           </property>
          </widget>
          <widget class="QLabel" name="baudRateLabel">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>50</y>
             <width>80</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>Baud Rate:</string>
           </property>
          </widget>
          <widget class="QLabel" name="parityLabel">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>90</y>
             <width>40</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>Parity:</string>
           </property>
          </widget>
          <widget class="QLabel" name="stopBitLabel">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>130</y>
             <width>60</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>Stop Bit:</string>
           </property>
          </widget>
          <widget class="QPushButton" name="selectDirectoryButton">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>170</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="text">
            <string>Select Directory</string>
           </property>
          </widget>
          <widget class="QLabel" name="directoryLabel">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>170</y>
             <width>150</width>
             <height>30</height>
            </rect>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QWidget" name="gaugeWidget" native="true">
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="1">
        <layout class="QGridLayout" name="torqueLayout"/>
       </item>
       <item row="2" column="1">
        <layout class="QGridLayout" name="vibrationLayout"/>
       </item>
       <item row="2" column="0">
        <layout class="QGridLayout" name="motorSpeedLayout"/>
       </item>
       <item row="1" column="0">
        <layout class="QGridLayout" name="fuelLayout"/>
       </item>
       <item row="0" column="0">
        <layout class="QGridLayout" name="oilPressureLayout"/>
       </item>
       <item row="0" column="1">
        <layout class="QGridLayout" name="oilTempLayout"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
